lua/nvim-treesitter/install.lua

```diff
@@ -37,6 +37,30 @@ local function get_revision(lang)
   return (lockfile[lang] and lockfile[lang].revision)
 end
 
+local function select_mkdir_cmd(directory, cwd, info_msg)
+  if fn.has('win32') == 1 then
+    return {
+      cmd = 'cmd',
+      opts = {
+        args = { '/C', 'mkdir', directory},
+	cwd = cwd,
+      },
+      info = info_msg,
+      err = "Could not create "..directory,
+    }
+  else
+    return {
+      cmd = 'mkdir',
+      opts = {
+        args = { directory },
+	cwd = cwd,
+      },
+      info = info_msg,
+      err = "Could not create "..directory,
+    }
+  end
+end
+
```

Comment:
you could also create this with `vim.fn.mkdir` beforehand. But this should be also ok. Could you move all these `select-*` functions to a separate file. `select_compiler_args` would also be a better name for `select_args`.

Reply:
Do you have a preference for the filename? Would command_selectors.lua be acceptable?

Reply:
`install_shell_commands.lua`?, `shell_command_selection.lua`?, `shell_command_selectors.lua`? I would add the `shell_` to bring some context which kind of commands it refers to.

Reply:
I've gone with `shell_command_selectors.lua`.

Reply:
> you could also create this with `vim.fn.mkdir` beforehand.
I tried using fn.mkdir for creating a new directory instead of defining a separate function for it.

But it didn't work since the newly created directory got deleted again by `M.select_install_rm_cmd(cache_folder, project_name..'-tmp')`.

Reply:
when you delete the `-tmp` directory before `mkdir` the first `select_install_rm_cmd` is not needed. But it should be ok as it is right now.